app/models/user.rb

```diff
@@ -62,4 +62,12 @@ def update!(user_params)
       filename: user_params[:photo_filename]
     )
   end
+
+  def self.from_social_provider(provider, user_params)
+    where(provider: provider, uid: user_params['id']).first_or_create do |user|
+      user.password = Devise.friendly_token[0, 20]
+      user.confirm
```

Comment:
maybe I got it wrong, but aren't we re-assigning a random password on every login with facebook? shouldn't we do it only the first time?